lib/phlex/context.rb

```diff
@@ -6,19 +6,45 @@ def initialize(user_context = {})
 		@buffer = +""
 		@capturing = false
 		@user_context = user_context
-		@fragment = nil
+		@fragments = nil
 		@in_target_fragment = false
-		@found_target_fragment = false
+		@halt_signal = nil
 	end
 
-	attr_accessor :buffer, :capturing, :user_context, :fragment,
-		:in_target_fragment, :found_target_fragment
+	attr_accessor :buffer, :capturing, :user_context, :in_target_fragment
+
+	attr_reader :fragments
 
 	# Added for backwards compatibility with phlex-rails. We can remove this with 2.0
 	def target
 		@buffer
 	end
 
+	def target_fragments(fragments)
+		@fragments = fragments.to_h { |it| [it, true] }
+	end
+
+	def around_render
+		return yield if @halt_signal
```

Comment:
We should check if `!fragments || @halt_signal` here.

Reply:
What's faster? `!fragments` or `@halt_signal`? I assume `@halt_signal` is the fastest. In fact, it may faster to skip the `fragments` check entirely, since that will run on every `around_render`. It's just there to prevent a `catch` that will only happen once any way.

Reply:
I guess `!@fragments` would be as fast as `@halt_signal`